lib/srv/db/server.go

```diff
@@ -99,14 +103,23 @@ func (c *Config) CheckAndSetDefaults() error {
 	if len(c.Servers) == 0 {
 		return trace.BadParameter("missing Servers")
 	}
-	if c.Credentials == nil {
+	// Only initialize AWS session if this service is proxying any RDS databases.
+	if c.AWSCredentials == nil && c.Servers.HasRDS() {
 		session, err := awssession.NewSessionWithOptions(awssession.Options{
 			SharedConfigState: awssession.SharedConfigEnable,
 		})
 		if err != nil {
 			return trace.Wrap(err)
 		}
-		c.Credentials = session.Config.Credentials
+		c.AWSCredentials = session.Config.Credentials
+	}
+	// Only initialize GCP IAM client if this service is proxying any Cloud SQL databases.
```

Comment:
How likely is it for a service to proxy both AWS RDS and Cloud SQL? Is this something users will potentially want to do?

Reply:
Don't know to be honest, probably not very likely in a real production setting, but it's a valid use-case and very convenient for development/testing - i.e. my db service is connected to everything.

Why, do you have something in mind for this particular block of code depending on whether this is a likely scenario?

Reply:
No, I was just thinking in terms of potential separation of concerns in db-specific auth implementation - currently it aggregates logic for each server type.